src/SuperVaultFactory.sol

```diff
@@ -0,0 +1,144 @@
+    ITokenizedStrategy public immutable tokenizedStrategy;
+
+    /// @notice The number of SuperVaults created
+    uint256 public superVaultCount;
+
+    /// @notice The total weight used for calculating proportions (10000 = 100%)
+    uint256 public constant TOTAL_WEIGHT = 10_000;
+
+    /// @notice The array of registered SuperVaults
+    address[] public superVaults;
+
+    /// @notice The mapping of registered SuperVaults
+    mapping(address superVault => bool registered) public registeredSuperVaults;
+
+    //////////////////////////////////////////////////////////////
+    //                       MODIFIERS                          //
+    //////////////////////////////////////////////////////////////
+
+    modifier onlyManagement() {
+        if (!hasRole(keccak256("MANAGEMENT_ROLE"), msg.sender)) {
+            revert NOT_MANAGEMENT();
+        }
+        _;
+    }
+
+    //////////////////////////////////////////////////////////////
+    //                       CONSTRUCTOR                        //
+    //////////////////////////////////////////////////////////////
+
+    /// @param superRegistry_ Address of the SuperRegistry
+    constructor(address superRegistry_) {
+        if (superRegistry_ == address(0)) {
+            revert ZERO_ADDRESS();
+        }
+        superRegistry = ISuperRegistry(superRegistry_);
+        _grantRole(keccak256("MANAGEMENT_ROLE"), msg.sender);
+    }
+
+    //////////////////////////////////////////////////////////////
+    //                    EXTERNAL WRITE FUNCTIONS              //
+    //////////////////////////////////////////////////////////////
+
+    /// @inheritdoc ISuperVaultFactory
+    /// @dev after deploying superVault, deployer needs to accept management
+    function createSuperVault(
+        address asset_,
+        address strategist_,
+        address vaultManager_,
+        string memory name_,
+        uint256 depositLimit_,
+        uint256[] memory superformIds_,
+        uint256[] memory startingWeights_
+    )
+        external
+        onlyManagement
+        returns (address)
+    {
+        if (asset_ == address(0) || strategist_ == address(0)) {
+            revert ZERO_ADDRESS();
+        }
+
+        uint256 numberOfSuperforms = superformIds_.length;
+
+        if (numberOfSuperforms == 0) {
+            revert ZERO_SUPERFORMS();
+        }
+
+        if (numberOfSuperforms != startingWeights_.length) {
+            revert ARRAY_LENGTH_MISMATCH();
+        }
+
+        bytes32 salt = keccak256(abi.encodePacked(asset_, name_, superformIds_, startingWeights_, "SuperVault"));
+
+        address superVault = address(
+            new SuperVault{ salt: salt }(
+                address(superRegistry),
+                asset_,
+                strategist_,
+                vaultManager_,
+                name_,
+                depositLimit_,
+                superformIds_,
+                startingWeights_
+            )
+        );
```

Comment:
what if a `supervault` is already created with the same parameters?
suggestion: check it doesn't exist yet

Reply:
We used `CREATE2` in the `createSuperVault()` function which reverts if a duplicate `SuperVault` is attempted to be created, this is tested [here](https://github.com/superform-xyz/SuperVaults/blob/yAudit/test/SuperVaultFactory.t.sol#L213)